packages/portfolio-contract/src/resolver/resolver.exo.ts

```diff
@@ -0,0 +1,212 @@
+  zcf: ZCF,
+  vowTools: VowTools,
+) => {
+  const cctpTransactionRegistry = resolverZone.mapStore<
+    CCTPTransactionKey,
+    CCTPTransactionEntry
+  >('cctpTransactionRegistry');
+
+  return resolverZone.exoClassKit(
+    'CCTPResolver',
+    {
+      client: ClientFacetI,
+      service: ServiceFacetI,
+      invitationMakers: InvitationMakersFacetI,
+    },
+    () => ({}),
+    {
+      client: {
+        /**
+         * Register a CCTP transaction for tracking.
+         *
+         * Note: Re-registering an already registered transaction is not allowed
+         * and returns the existing vow for that transaction.
+         *
+         * @param destinationAddress - The destination address for the transaction
+         * @param amountValue - The amount value for the transaction
+         * @returns A vow that resolves when the transaction is settled
+         */
+        registerCCTPTransaction(
+          destinationAddress: AccountId,
+          amountValue: NatValue,
+        ) {
+          const key =
+            `${destinationAddress}:${amountValue}` as CCTPTransactionKey;
+
+          if (cctpTransactionRegistry.has(key)) {
+            trace(`CCTP transaction already registered: ${key}`);
+            return cctpTransactionRegistry.get(key).vowKit.vow;
+          }
+
+          const vowKit = vowTools.makeVowKit<void>();
+
+          cctpTransactionRegistry.init(
+            key,
+            harden({
+              destinationAddress,
+              amountValue: amountValue,
+              vowKit,
+            }),
+          );
+
+          trace(`Registered pending CCTP transaction: ${key}`);
+          return vowKit.vow;
+        },
+      },
+      service: {
+        settleCCTPTransaction(
+          chainId: CaipChainId,
+          remoteAddress: `0x${string}`,
+          amountValue: bigint,
+          status: 'confirmed' | 'failed' | 'pending',
+        ) {
+          const key =
+            `${chainId}:${remoteAddress}:${amountValue}` as CCTPTransactionKey;
+
+          if (!cctpTransactionRegistry.has(key)) {
+            trace('No pending CCTP transaction found for key:', key);
+            trace('Available keys in registry:', [
+              ...cctpTransactionRegistry.keys(),
+            ]);
+            throw assert.error(
+              X`No pending CCTP transaction found matching provided details: ${q(key)}`,
+            );
+          }
+
+          const registryEntry = cctpTransactionRegistry.get(key);
+
+          switch (status) {
+            case 'confirmed':
+              trace(`CCTP transaction confirmed - resolving pending operation`);
+              registryEntry.vowKit.resolver.resolve();
+              cctpTransactionRegistry.delete(key);
+              return;
+
+            case 'failed':
+              trace(`CCTP transaction failed - rejecting pending operation`);
+              registryEntry.vowKit.resolver.reject(
+                assert.error('CCTP transaction failed'),
+              );
```

Comment:
I find this use of `assert` misleading.
```suggestion
              registryEntry.vowKit.resolver.reject(
                Error('CCTP transaction failed'),
              );
```